core/src/test/scala/za/co/absa/spline/harvester/builder/read/ReadCommandExtractorSpec.scala

```diff
@@ -0,0 +1,65 @@
+package za.co.absa.spline.harvester.builder.read
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.plans.logical.{LeafNode, LogicalPlan, Statistics}
+import org.apache.spark.sql.execution.datasources.LogicalRelation
+import org.apache.spark.sql.sources.BaseRelation
+import org.apache.spark.sql.types.StructType
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatestplus.mockito.MockitoSugar
+import za.co.absa.commons.spark.SparkTestBase
+import za.co.absa.spline.harvester.builder.SourceIdentifier
+import za.co.absa.spline.harvester.qualifier.HDFSPathQualifier
+
+import scala.util.{Failure, Try}
+
+class ReadCommandExtractorSpec extends AnyFlatSpec
+                                       with Matchers
+                                       with SparkTestBase
+                                       with MockitoSugar {
+
+  private val conf = new Configuration()
+  private val pathQualifier = new HDFSPathQualifier(conf)
+
+  it must "defer to the relation handler if there is a compatible one" in {
+    val logicalPlan: TestLogicalPlan = new TestLogicalPlan(TestRelation(spark.sqlContext))
+    object TestRelationHandler extends ReadRelationHandler {
+      override def isApplicable(relation: BaseRelation): Boolean = relation.isInstanceOf[TestRelation]
+
+      override def apply(relation: BaseRelation,
+                         logicalPlan: LogicalPlan): ReadCommand = ReadCommand(SourceIdentifier(Some("test")),
+                                                                              logicalPlan)
+    }
+
+    val result: Option[ReadCommand] = new ReadCommandExtractor(pathQualifier,
+                                                               spark,
+                                                               TestRelationHandler)
+      .asReadCommand(logicalPlan)
+    result.isDefined mustBe true
+    result.map(rc => rc.sourceIdentifier).flatMap(si => si.format).getOrElse("fail") mustBe "test"
+  }
+
+  it must "return a None if there is no way to handle the relation" in {
+    val relationNotHandled: Boolean = Try(new ReadCommandExtractor(pathQualifier,
+                                                                   spark)
+                                            .asReadCommand(new TestLogicalPlan(TestRelation(spark.sqlContext)))) match {
+      case _: Failure[_] => true
+      case _ => false
+    }
```

Comment:
```suggestion
    val relationNotHandled: Boolean = 
      Try(
        new ReadCommandExtractor(pathQualifier,spark)
          .asReadCommand(newTestLogicalPlan(TestRelation(spark.sqlContext)))
      ) 
      match {
        case _: Failure[_] => true
        case _ => false
      }
```

This is not really formatting style that we use in the project. It would be nice to keep the style consistent.
I tried to format the code as example, but you can use at the rest of the code or/and take a look at indentation chapter in this Scala style guide: https://github.com/databricks/scala-style-guide#indent

Please reformat the rest as well so it's consistent with our code style.

Reply:
Thanks, I'll review the code with the code style